cli/pkg/schema/validate.go

```diff
@@ -8,12 +8,28 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
+func (a *Application) checkNoEnvVarCollisions() []gojsonschema.ResultError {
```

Comment:
Should this validation also apply during edit? It doesn't at the moment.

Reply:
Looks like the validation on file edits ONLY ever runs if there is actually an error loading the yaml config. See filesync.go. This was to prevent broken config from reaching the UI. I'll raise a new issue to have additional validations included as part of that as currently none of them are because of this case.

I've tested it out locally with some changes but it has a much bigger impact on UX (completely blocking the user from using the UI at all! So I think this warrants a separate issue.